flaky test failure in RegistryDelete

I ran into this yesterday in a local build - re-running "fixed it" - and @bboreham [encountered it in CI today](https://circleci.com/gh/weaveworks/scope/8868#tests/containers/1)
```
--- FAIL: TestRegistryDelete (0.01s)
	registry_test.go:511: Didn't get right container updates: 
		--- want
		+++ have
		@@ -1,4 +1,322 @@
		-([]report.Node) (len=1 cap=1) {
		+([]report.Node) (len=2 cap=2) {
                ...
```
Presumably a race condition in the test rather than a bug in the code.

--- a/probe/docker/registry_test.go
+++ b/probe/docker/registry_test.go
@@ -470,7 +470,7 @@ func TestRegistryDelete(t *testing.T) {
 	setupStubs(mdc, func() {
 		registry := testRegistry()
 		defer registry.Stop()
-		runtime.Gosched()
+		time.Sleep(time.Millisecond * 100) // Allow for goroutines to get started
 
 		// Collect all the events.
 		mtx := sync.Mutex{}

